src/MultiFormats/MultiBase.cs

```diff
@@ -0,0 +1,123 @@
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using MultiFormats.Registry;
+
+namespace MultiFormats
+{
+    /// <summary>
+    ///   Self identifying base encodings.
+    /// </summary>
+    /// <remarks>
+    ///   <b>MultiBase</b> is a protocol for distinguishing base encodings 
+    ///   and other simple string encodings.  
+    ///   See the <see cref="MultiBaseAlgorithm">registry</see> for supported algorithms.
+    /// </remarks>
+    /// <seealso href="https://github.com/multiformats/multibase"/>
+    public static class MultiBase
+    {
+        /// <summary>
+        ///   The default multi-base algorithm is "base58btc".
+        /// </summary>
+        public const string DefaultAlgorithmName = "base58btc";
+
+        /// <summary>
+        ///   Gets the <see cref="MultiBaseAlgorithm"/> with the specified IPFS multi-hash name.
+        /// </summary>
+        /// <param name="name">
+        ///   The name of an algorithm, see 
+        ///   <see href="https://github.com/multiformats/multibase/blob/master/multibase.csv"/> for
+        ///   for IPFS defined names.
+        /// </param>
+        /// <exception cref="KeyNotFoundException">
+        ///   When <paramref name="name"/> is not registered.
+        /// </exception>
+        private static MultiBaseAlgorithm GetAlgorithm(string name)
+        {
+            try
+            {
+                return MultiBaseAlgorithm.Names[name];
+            }
+            catch (KeyNotFoundException)
+            {
+                throw new KeyNotFoundException($"MutiBase algorithm '{name}' is not registered.");
+            }
+        }
+
+        /// <summary>
+        ///   Converts an array of 8-bit unsigned integers to its equivalent string representation.
+        /// </summary>
+        /// <param name="bytes">
+        ///   An array of 8-bit unsigned integers.
+        /// </param>
+        /// <param name="algorithmName">
+        ///   The name of the multi-base algorithm to use. See <see href="https://github.com/multiformats/multibase/blob/master/multibase.csv"/>.
+        /// </param>
+        /// <returns>
+        ///   A <see cref="string"/> starting with the algorithm's <see cref="MultiBaseAlgorithm.Code"/> and
+        ///   followed by the encoded string representation of the <paramref name="bytes"/>.
+        /// </returns>
+        /// <exception cref="KeyNotFoundException">
+        ///   When <paramref name="algorithmName"/> is not registered.
+        /// </exception>
+        public static string Encode(byte[] bytes, string algorithmName = DefaultAlgorithmName)
+        {
+            if (bytes == null) throw new ArgumentNullException("bytes");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)